app/domain/authentication/authn_jwt/input_validation/parse_mandatory_claims.rb

```diff
@@ -0,0 +1,97 @@
+      ParseMandatoryClaims ||= CommandClass.new(
+        dependencies: {
+          validate_claim_name: ValidateClaimName.new(
+            deny_claims_list_value: MANDATORY_CLAIMS_DENY_LIST
+          ),
+          logger: Rails.logger
+        },
+        inputs: %i[mandatory_claims_secret_value]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsingMandatoryClaims.new(@mandatory_claims_secret_value))
+          validate_mandatory_claims_secret_value_exists
+          validate_mandatory_claims_list_format
+          validate_mandatory_claims_list_value
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsedMandatoryClaims.new(parsed_mandatory_claims_list))
+
+          parsed_mandatory_claims_list
+        end
+
+        private
+
+        def validate_mandatory_claims_secret_value_exists
+          raise Errors::Authentication::AuthnJwt::FailedToParseMandatoryClaimsMissingInput if @mandatory_claims_secret_value.blank?
+        end
+        
+        def validate_mandatory_claims_list_format
+          validate_delimiter_format
+          validate_duplications
+          validate_claims_names_without_spaces
+        end
+
+        def validate_delimiter_format
+          if mandatory_claims_secret_value_starts_or_ends_with_delimiter? || 
+              mandatory_claims_secret_value_has_connected_delimiter?
+            raise Errors::Authentication::AuthnJwt::InvalidMandatoryClaimsFormat, @mandatory_claims_secret_value
+          end
+        end
+        
+        def mandatory_claims_secret_value_starts_or_ends_with_delimiter?
+          mandatory_claims_secret_value_first_character == CLAIMS_CHARACTER_DELIMITER ||
+            mandatory_claims_secret_value_last_character == CLAIMS_CHARACTER_DELIMITER
+        end
+          
+        def mandatory_claims_secret_value_first_character
+          @mandatory_claims_secret_value_first_character ||= @mandatory_claims_secret_value[0, 1]
+        end
+
+        def mandatory_claims_secret_value_last_character
+          @mandatory_claims_secret_value_last_character ||= @mandatory_claims_secret_value[-1]
+        end
+        
+        def mandatory_claims_secret_value_has_connected_delimiter?
+          parsed_mandatory_claims_list.include?('')
+        end
+
+        def validate_duplications
+          if parsed_mandatory_claims_list.uniq.length != parsed_mandatory_claims_list.length
+            raise Errors::Authentication::AuthnJwt::InvalidMandatoryClaimsFormatContainsDuplication, @mandatory_claims_secret_value
+          end
+        end
+        
+        def parsed_mandatory_claims_list
+          @parsed_mandatory_claims_list || mandatory_claims_secret_value_trim.split(CLAIMS_CHARACTER_DELIMITER)
+        end
+        
+        def mandatory_claims_secret_value_trim
+          @mandatory_claims_secret_value_trim ||= @mandatory_claims_secret_value.delete(' ')
+        end
+
+        def validate_claims_names_without_spaces
```

Comment:
Dont you check it already in `mandatory_claims_secret_value_has_connected_delimiter?`?

Reply:
look a the UTs, in this step i will fail the following input: `claim  _name1, claim_name2`